web/src/stream_events.ts

```diff
@@ -177,6 +181,43 @@ export function update_property<P extends keyof UpdatableStreamProperties>(
             update_stream_setting(sub, value, "is_recently_active");
             stream_list.update_streams_sidebar();
         },
+        is_archived(value) {
+            if (!value) {
+                // We currently do not live-udpate when unarchiving stream.
+                return;
+            }
+            const is_subscribed = sub.subscribed;
+            const is_narrowed_to_stream = narrow_state.narrowed_to_stream_id(stream_id);
+            stream_data.mark_archived(stream_id);
+            stream_settings_ui.update_settings_for_archived(sub);
+            message_view_header.maybe_rerender_title_area_for_stream(stream_id);
+            if (is_subscribed) {
+                stream_list.remove_sidebar_row(stream_id);
+                if (stream_id === compose_state.selected_recipient_id) {
+                    compose_state.set_selected_recipient_id("");
+                    compose_recipient.on_compose_select_recipient_update();
+                }
+            }
+            settings_streams.update_default_streams_table();
+            stream_data.remove_default_stream(stream_id);
+            if (realm.realm_new_stream_announcements_stream_id === stream_id) {
+                realm.realm_new_stream_announcements_stream_id = -1;
+                settings_org.sync_realm_settings("new_stream_announcements_stream_id");
+            }
+            if (realm.realm_signup_announcements_stream_id === stream_id) {
+                realm.realm_signup_announcements_stream_id = -1;
+                settings_org.sync_realm_settings("signup_announcements_stream_id");
+            }
+            if (realm.realm_zulip_update_announcements_stream_id === stream_id) {
+                realm.realm_zulip_update_announcements_stream_id = -1;
+                settings_org.sync_realm_settings("zulip_update_announcements_stream_id");
+            }
```

Comment:
I'm a bit surprised we need this code; `maybe_set_moderation_or_announcement_channels_none` should be sending the events to cause those updates to happen.

Reply:
Yes, we do not need this. Removed.